ape_starknet/accounts/__init__.py

```diff
@@ -629,7 +631,9 @@ def prepare_transaction(self, txn: TransactionAPI) -> TransactionAPI:
 
         txn.max_fee = txn.max_fee or MAX_FEE
         txn = self._prepare_transaction(txn)
-        txn.signature = self.sign_transaction(txn)
+        signed_txn = self.sign_transaction(txn)
+        if signed_txn is not None:
+            txn = cast(TransactionAPI, txn)
```

Comment:
```suggestion
            return cast(TransactionAPI, signed_txn)
```

You are going to want to return the signed txn here